util/reward-calculator/src/lib.rs

```diff
@@ -0,0 +1,194 @@
+use ckb_core::header::{BlockNumber, Header};
+use ckb_core::script::Script;
+use ckb_core::transaction::ProposalShortId;
+use ckb_core::Capacity;
+use ckb_store::ChainStore;
+use ckb_traits::ChainProvider;
+use failure::{Error as FailureError, Fail};
+use fnv::FnvHashSet;
+use numext_fixed_hash::H256;
+use std::cmp;
+use std::collections::BTreeMap;
+
+#[derive(Debug, PartialEq, Clone, Eq, Fail)]
+pub enum Error {
+    #[fail(display = "Can't resolve finalize target: {}", _0)]
+    Target(BlockNumber),
+    #[fail(display = "Can't parse Script from target witness: {:x}", _0)]
+    Script(H256),
+}
+
+pub struct RewardCalculator<'a, P> {
```

Comment:
I think we can add some ascii chart here to describe how the reward is calculated.